src/api/app/controllers/services/webhooks_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Services::WebhooksController < ApplicationController
+  skip_before_action :extract_user
+  skip_before_action :require_login
+  skip_before_action :validate_params
+  before_action :validate_token, :set_package, :set_user
+
+  def create
+    if !@user.is_active? || !@user.can_modify?(@package)
+      render_error message: 'Token not found or not valid.', status: 404
+      return
+    end
+
+    Backend::Api::Sources::Package.trigger_services(@project.name, @package.name, @user.login)
+    render_ok
+  end
+
+  private
+
+  def set_package
+    @package = @token.package || Package.get_by_project_and_name(params[:project], params[:package], use_source: true)
+    @project = @package.project
```

Comment:
I would move this to line 13 as well. It doesn't fit into a 'set_package' method and we don't gain much from having it here.

Reply:
Done